docker/transport/sshconn.py

```diff
@@ -90,8 +104,22 @@ def __init__(self, base_url, timeout=60,
 
     def _connect(self):
         parsed = six.moves.urllib_parse.urlparse(self.base_url)
```

Comment:
With the addition above we are not now parsing the URL twice? We should parse it exactly once. That suggests either the code here should be in `__init__` or the code from `__init__` should be here.

Reply:
I tried to do rather minimal refactoring. I changed this now consistently.

After looking at the code I however wonder, why a connection is created at all in the init code and why base_url is passed to the constructor...